### PR TITLE
My Library JS Project

### PR DESCRIPTION
**Pull request by SmithVyne**
We created a simple Javascript application where users can:

1. Add new books they know to the library with a form.
2. Select whether or not they have read the books.
3. Update the statuses of the books later if they read them.
4. Delete books from the Library.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# My Library
+# JS-Library-Project at Microverse
+Built-with: Javascript, HTML, CSS + Bootstrap
+
+# Set-Up instructions.
+* Clone this repository by running `https://github.com/SmithVyne/JS-Library-Project.git` and cd into the directory.
+* Open `index.html` in the browser and interact with the application.
```
Comment (lekegitrepo) [inline README.md:7]: - **_The readme should contain information about installing the dependencies._**

File: script.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          <p class="card-text book-author">${book.author}</p>
+          <p class="card-text book-pages">${book.pages}</p>
+          <p class="card-text d-flex box-read">
+          <span> Read: ${hasRead(book, i)} </span>
+          </p>
+          <button class="btn btn-danger" onclick="deleteBook(${i})">Delete</button>
+        </div>
+      </div>`;
+  }
+};
```
Comment (lekegitrepo) [inline script.js:62]: - **_The Display Book should have a `render` function ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/render_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library))_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
### Good job so far :muscle: However, you need to implement the following requirements before this project can be approved.

- **_There should a package.json(I am not talking about the package-lock.json) file in your project which should contain all the dependencies_**

### OPTIONAL
* [ ] The form used to create a book should have validations, you should not be able to create an empty book
* [ ] It has data persistence by saving, retrieving, and updating the library books using [the localstorage API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/localstorage_stretch_requirement.png) from [Odin Project](https://www.theodinproject.com/courses/javascript/lessons/library))

#### _Reviewed by Temitayo Adeleke._

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **SmithVyne Nkereuwem & Alfred Ezaka**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with this project, it works quite well. There are some issues with the code organization and the linting but the nailed the main functionality of it!

Rock on! 🎸 👨‍🎤 

# Required Changes 🏗️ 🐛 💡

- [x] Please fix the eslint errors you are getting in the PR checks.
Screenshot_2020-09-16 My Library JS Project by SmithVyne · Pull Request #2 · SmithVyne JS-Library-Project

- [x] [OPTIONAL] You can add data persistence by saving, retrieving, and updating the library books using [the localstorage API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API).  ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/localstorage_stretch_requirement.png) from [Odin Project](https://www.theodinproject.com/courses/javascript/lessons/library))
That way the books you add will still be there even after you refresh the page.

(Check the comments below for the other fixes)

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @SmithVyne and @codecell 
### You have done well so far :clap: However, a few more issues are required for you to fix:-
- [ ] Host the app online and share the link on the readme file
- [ ] Your linting is producing error, kindly fi this. See the picture below
lint
- [ ] Separate app logic from dom manipulation
### Make these changes and submit for another code review. Good luck.
- Happy hacking!

By Raymond
Slack @iamraymond

File: script.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        <p class="card-text book-author">${book.author}</p>
+        <p class="card-text book-pages">${book.pages}</p>
+        <p class="card-text d-flex box-read">
+        <span> Read: ${hasRead(book, i)} </span>
+        </p>
+        <button class="btn btn-danger" onclick="deleteBook(${i})">Delete</button>
+      </div>
+    </div>`;
+  }
+};
```
Comment (rOluochKe) [inline script.js:76]: ### Separate app dom manipulation from app logic

File: script.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+function addBook() {
+  const btnAddbook = document.getElementById('btnAddbook');
+
+  btnAddbook.addEventListener('click', () => {
+    const title = document.getElementById('title');
+    const author = document.getElementById('author');
+    const pages = document.getElementById('pages');
+    const box = document.querySelector('#check-book');
```
Comment (rOluochKe) [inline script.js:105]: ### Separate app dom manipulation from app logic

**Comment by SmithVyne** [comment]
Hello @rOluochKe , hosting the app online is not part of the requirements

File: script.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  } else if (book.title === '') {
+    errors.push('Book title cannot be empty');
+  } else if (book.pages === '') {
+    errors.push('Number of Pages cannot be empty');
+  }
+
+  return errors;
+};
+
+const hasRead = (book, i) => {
```
[the remark this answers is not in the document]
Comment (SmithVyne) [reply script.js:48]: This would cause lint errors because the functions would be used before they were declared

File: script.js (hunk 3 above)
Comment (SmithVyne) [reply script.js:76]: This is a loop, and the best way to implement this section. Any other would be using more lines of code to do what we can in less

**Comment by sumancrest0001** [comment]
**Status: required changes :gear:
Hello TEAM,** You have done a nice job so far and make the website working as expected. But there are a couple of things that need to be improved. As mentioned by @uryelah  and @rOluochKe, we need to separate the logical part from the DOM part. It will make the code cleaner and file manageable. Let's not forget that we have to implement ES6 whenever it is applicable.

File: script.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+function Book(title, author, pages, read) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.read = read;
+
+  this.info = function () {
+    return `${title} ${author} ${pages} ${read}`;
+  };
```
Comment (sumancrest0001) [inline script.js:32]: [required] It seems that you haven't used this function. If not, please remove it.

File: script.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (sumancrest0001) [reply script.js:35]: Yes you are right.

File: script.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+};
+
+function addBook() {
+  const title = document.getElementById('title');
+  const author = document.getElementById('author');
+  const pages = document.getElementById('pages');
+  const box = document.querySelector('#check-book');
+  const read = !!box.checked;
+
+  const newBook = new Book(title.value, author.value, pages.value, read);
```
Comment (sumancrest0001) [inline script.js:130]: [Required] Let's create a separate function to get values from input fields and return then and use those values to create a new book.

File: script.js [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+    author: 'Paul West',
+    title: 'John\'s travels',
+    pages: 200,
+    read: true,
+  },
+];
+
+const library = [...dummyBooks];
+
+function Book(title, author, pages, read) {
```
Comment (sumancrest0001) [inline script.js:24]: [Required] Let's use ES6.

File: script.js [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+const changeTrue = (i) => {
+  library[i].read = false;
+  displayBooks(library);
+};
+
+const changeFalse = (i) => {
+  library[i].read = true;
+  displayBooks(library);
+};
+
```
Comment (sumancrest0001) [inline script.js:77]: [Required] You do not need parentheses for a single argument.
[Optional] You can use something meaningful instead of i.

**Review by sumancrest0001 — CHANGES_REQUESTED** [review]
**Kindly make the requested changes and submit the project for another review.
If you have any questions, Dm me on Slack (Suman Shrestha).
Happy coding !!!**

File: script.js (hunk 6 above)
Comment (SmithVyne) [reply script.js:32]: The info method is part of the instructions for the Book constructor class

File: script.js (hunk 2 above)
Comment (SmithVyne) [reply script.js]: I'm afraid that is an old requirement for this project [Odin](https://www.theodinproject.com/courses/javascript/lessons/library)

Comment (SmithVyne) [reply script.js]: This is outdated requirement for this project [Odin link](https://www.theodinproject.com/courses/javascript/lessons/library)

File: script.js (hunk 8 above)
Comment (SmithVyne) [reply script.js:24]: Okay

File: script.js (hunk 7 above)
Comment (SmithVyne) [reply script.js:130]: That's possible, @sumancrest0001, but I do not think it is worth the extra lines of code.

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED

Hi @SmithVyne  @codecell nice job working on this team, this looks really good to me, everything works fine, kudos for making changes. **This is approved** 

Please merge and move on :slightly_smiling_face: 

**Raphael Noriode** :mask: STAY_SAFE

**Pull request merged**